src/librustc_codegen_llvm/build.rs

```diff
@@ -8,9 +8,273 @@
+        } else {
+            continue;
+        };
+
+        // Don't need or want this library, but LLVM's CMake build system
+        // doesn't provide a way to disable it, so filter it here even though we
+        // may or may not have built it. We don't reference anything from this
+        // library and it otherwise may just pull in extra dependencies on
+        // libedit which we don't want
+        if name == "LLVMLineEditor" {
+            continue;
+        }
+
+        let kind = if name.starts_with("LLVM") {
+            llvm_kind
+        } else {
+            "dylib"
+        };
+        println!("cargo:rustc-link-lib={}={}", kind, name);
+    }
+
+    // LLVM ldflags
+    //
+    // If we're a cross-compile of LLVM then unfortunately we can't trust these
+    // ldflags (largely where all the LLVM libs are located). Currently just
+    // hack around this by replacing the host triple with the target and pray
+    // that those -L directories are the same!
+    let mut cmd = Command::new(&llvm_config);
+    cmd.arg(llvm_link_arg).arg("--ldflags");
+    for lib in output(&mut cmd).split_whitespace() {
+        if lib.starts_with("-LIBPATH:") {
+            println!("cargo:rustc-link-search=native={}", &lib[9..]);
+        } else if is_crossed {
+            if lib.starts_with("-L") {
+                println!("cargo:rustc-link-search=native={}",
+                         lib[2..].replace(&host, &target));
+            }
+        } else if lib.starts_with("-l") {
+            println!("cargo:rustc-link-lib={}", &lib[2..]);
+        } else if lib.starts_with("-L") {
+            println!("cargo:rustc-link-search=native={}", &lib[2..]);
+        }
+    }
+
+    let llvm_static_stdcpp = env::var_os("LLVM_STATIC_STDCPP");
+
+    let stdcppname = if target.contains("openbsd") {
+        // llvm-config on OpenBSD doesn't mention stdlib=libc++
+        "c++"
+    } else if target.contains("freebsd") {
+        "c++"
+    } else if target.contains("netbsd") && llvm_static_stdcpp.is_some() {
+        // NetBSD uses a separate library when relocation is required
+        "stdc++_pic"
+    } else {
+        "stdc++"
+    };
+
+    // C++ runtime library
+    if !target.contains("msvc") {
+        if let Some(s) = llvm_static_stdcpp {
+            assert!(!cxxflags.contains("stdlib=libc++"));
+            let path = PathBuf::from(s);
+            println!("cargo:rustc-link-search=native={}",
+                     path.parent().unwrap().display());
+            println!("cargo:rustc-link-lib=static={}", stdcppname);
+        } else if cxxflags.contains("stdlib=libc++") {
+            println!("cargo:rustc-link-lib=c++");
+        } else {
+            println!("cargo:rustc-link-lib={}", stdcppname);
+        }
+    }
+
+    // LLVM requires symbols from this library, but apparently they're not printed
+    // during llvm-config?
+    if target.contains("windows-gnu") {
+        println!("cargo:rustc-link-lib=gcc_s");
```

Comment:
@alexcrichton I read in the Cargo docs that this will default to `dylib`, is that not the case? Or would that not help at all and we need to go back to having at least one extra staticlib?

Reply:
Ah this is informed to rustc that it's a "dylib" but it's not actually on the filesystem, it's a static library, so it ends up not really mattering.

I think a hack for now is to leave `librustc_llvm` as a husk which just links to `gcc_s` on MinGW and we can figure it out later

Reply:
Interesting, searching online gives me `libgcc_s.so` for the `emutls` symbols, I guess maybe mingw only has the static version? And yeah, @irinagpopa is going to install mingw (because getting the husk right might take a few tries, might as well not stress the bots further) and try the husk approach.

Reply:
The error message above refers to `C:/projects/rust/.../libgcc_s.a`, so I think on MinGW it's available as an archive instead of a static library